cfme/common/provider_views.py

```diff
@@ -250,3 +251,148 @@ def is_displayed(self):
     sidebar = View.nested(ProviderSideBar)
     entities = View.nested(ProviderEntities)
     paginator = View.nested(PaginationPane)
+
+
+class BeforeFillMixin(object):
+    def before_fill(self):
+        if self.exists and not self.is_active():
+            self.select()
+
+
+class DefaultEndpointForm(View):
```

Comment:
**REQUIRED:** These too should be moved into the provider files, then they will sit with their object counterparts